pilot/pkg/networking/core/v1alpha3/listener_builder.go

```diff
@@ -203,6 +213,24 @@ func (builder *ListenerBuilder) getListeners() []*xdsapi.Listener {
 	return builder.gatewayListeners
 }
 
+func newBlackhole(node *model.Proxy) *listener.Filter {
+	tcpProxy := &tcp_proxy.TcpProxy{
```

Comment:
Nit. I don’t think you need to allocate again and again, esp when you assign by value. Just have a constant global and copy it over.

Reply:
@howardjohn thoughts on this? Seems like you could even pre-compute the entire filters (i.e. pre-marshal to both any and struct)

Reply:
will change this to premarshal